From 7abaaf8a3cd6d5897a4323686fb2130433c2e662 Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Thu, 15 Aug 2019 11:38:16 +0200
Subject: [PATCH 1/2] e6b3ba4 dh_testroot: root_requirements() no longer read
 d/control

---
 debian/rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/rules b/debian/rules
index e020f7f..1304204 100755
--- a/debian/rules
+++ b/debian/rules
@@ -3,7 +3,7 @@ include /usr/share/dpkg/pkg-info.mk
 pod2man := pod2man -r "dh-autoreconf v$(DEB_VERSION)" -c dh-autoreconf
 
 %:
-	dh $@
+	DEB_RULES_REQUIRES_ROOT=no dh $@ --without autoreconf
 
 override_dh_auto_build:
 	$(pod2man) --section=1 dh_autoreconf dh_autoreconf.1
-- 
2.23.0.rc1

