From a28fd38a2617c32ad4ff9d0a989414e3f23d459c Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Thu, 15 Aug 2019 12:06:20 +0200
Subject: [PATCH 2/4] e6b3ba4 dh_testroot: root_requirements() no longer read
 d/control

---
 debian/rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/rules b/debian/rules
index 026823b..7e5a355 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1,7 +1,7 @@
 #!/usr/bin/make -f
 
 %:
-	dh $@
+	DEB_RULES_REQUIRES_ROOT=no dh $@ --without autoreconf
 
 # Remove the auto-generated Makefile to prevent dh_auto_clean from failing
 # if the Makefile contain paths referring to an old version of Perl.
-- 
2.23.0.rc1

