From 788790d1bf954dcf51aa1fd7be157ad89eb4c332 Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Sun, 28 Jul 2019 00:54:49 +0200
Subject: [PATCH 3/4] doc

---
 perl/Debian/DocBase/DocBaseFile.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/perl/Debian/DocBase/DocBaseFile.pm b/perl/Debian/DocBase/DocBaseFile.pm
index f7dd50f..b921b4c 100644
--- a/perl/Debian/DocBase/DocBaseFile.pm
+++ b/perl/Debian/DocBase/DocBaseFile.pm
@@ -438,7 +438,7 @@ sub _CheckRequiredFields($$$) { # {{{
 
     if (not grep { $_ eq $format } @SUPPORTED_FORMATS) {
       $self->_PrsErr(PRS_WARN, _g("format `%s' is not supported"), $$format_data{'format'});
-      next;
+      #next;
     }
 
     my $index_value = undef;
-- 
2.23.0.rc1

