From 65e2400a5c136dd346c82a4ba1838e634cf5a65b Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Wed, 31 Jul 2019 12:19:58 +0200
Subject: [PATCH 10/15] cgi

---
 perl/Debian/Dwww/ConfigFile.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/perl/Debian/Dwww/ConfigFile.pm b/perl/Debian/Dwww/ConfigFile.pm
index 5736b71..a46aedc 100644
--- a/perl/Debian/Dwww/ConfigFile.pm
+++ b/perl/Debian/Dwww/ConfigFile.pm
@@ -43,7 +43,7 @@ sub ReadConfigFile($) {
         },
         'DWWW_CGIUSER'  => {
             sort   => 10,
-            defval => 'www-data',
+            defval => 'Guest',
             descr  => 'Name of the user the web server uses to execute CGI script.'
         },
         # Browser variables
-- 
2.23.0.rc1

