From dd2bbdfa01e50afccab804b80a99ea7a9c5f00a9 Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Fri, 9 Aug 2019 20:39:22 +0200
Subject: [PATCH 15/15] distributor

---
 common.mk | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/common.mk b/common.mk
index a912b2f..cef4318 100644
--- a/common.mk
+++ b/common.mk
@@ -56,11 +56,7 @@ ifndef VERSION
   export VERSION DATE DATE_EN
 
   ifndef DISTRIBUTOR
-    ifneq (,$(findstring ubuntu,$(VERSION)))
-      DISTRIBUTOR     := Ubuntu
-    else
       DISTRIBUTOR     := `lsb_release -is`
-    endif
   endif
   distributor     := $(shell echo $(DISTRIBUTOR) | tr "[:upper:]" "[:lower:]")
 
-- 
2.23.0.rc1

