--- cygport-0.25.0/Makefile.am
+++ cygport-0.25.0/Makefile.am
@@ -21,6 +21,7 @@ EXTRA_DIST +=				\
 	cygport.spec			\
 	$(NULL)
 
+if HAVE_HELP2MAN
 dist-hook: doc/manual/cygport.1.html doc/manual/index.html
 	@if test -d "$(srcdir)/.git"; \
 	then \
@@ -35,3 +36,5 @@ dist-hook: doc/manual/cygport.1.html doc/manual/index.html
 		echo A git clone is required to generate a ChangeLog >&2; \
 	fi ; \
 	cp -r doc/manual/ $(top_distdir)/doc/
+else !HAVE_HELP2MAN
+endif !HAVE_HELP2MAN
--- cygport-0.25.0/cygclass/Makefile.am
+++ cygport-0.25.0/cygclass/Makefile.am
@@ -207,6 +207,7 @@ cygclass_DATA =					\
 	cygclass/distutils.cygclass		\
 	cygclass/distutils-multi.cygclass	\
 	cygclass/docbook.cygclass		\
+	cygclass/dpkg.cygclass		\
 	cygclass/ebook.cygclass			\
 	cygclass/emacs.cygclass			\
 	cygclass/font.cygclass			\
--- /dev/null
+++ cygport-0.25.0/cygclass/dpkg.cygclass
@@ -0,0 +1,73 @@
+################################################################################
+#
+# dpkg.cygclass - functions for building Debian packages
+#
+# Part of cygport - Cygwin packaging application
+# Copyright (C) 2006-2016 Yaakov Selkowitz
+# Provided by the Cygwin Ports project <http://sourceware.org/cygwinports/>
+#
+# cygport is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cygport is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cygport.  If not, see <http://www.gnu.org/licenses/>.
+#
+################################################################################
+
+CYGPORT_USE_UNSTABLE_API=1
+
+# cross-compiling is not (yet?) supported
+__cross_compiling_error
+
+check_prog_req dpkg-source
+check_prog_req dpkg-buildpackage
+
+src_unpack_hook(){
+	find -mindepth 1 -maxdepth 1 -type d |
+	xargs -I@ rm -rf @
+
+	basename * |
+	PERL5LIB=/usr/share/perl5 \
+	xargs -I@ dpkg-source --extract "${top}/@" "${NAME}"
+	sed -i "${NAME}/debian/control" \
+		-e /Rules-Requires-Root/d \
+		-e /^Source:/,/^$/s/^\$/Rules-Requires-Root:\ no/ \
+		-e T\;G
+}
+
+src_patch_hook(){
+	:;
+}
+
+#****o* dpkg.cygclass/src_compile (dpkg)
+#  DEFINITION
+src_compile() {
+	rsync --archive "${S}/." "${B}/." &&
+	cd -P "${B}/${NAME}" &&
+	PATH="$PWD/bin:$PATH" \
+	PERL5LIB="$PWD/lib:/usr/share/perl5" \
+	PERL_MM_OPT="INSTALLDIRS=vendor INSTALLVENDORLIB=/usr/share/perl5" \
+	dpkg-buildpackage -rnice -d -a i386 -t i686-pc-cygwin --source-option=--auto-commit
+}
+#****
+
+#****o* dpkg.cygclass/src_test (dpkg)
+#  DEFINITION
+src_test() {
+	cd -P "${B}"
+}
+#****
+
+#****o* dpkg.cygclass/src_install (dpkg)
+#  DEFINITION
+src_install() {
+	cd -P "${B}" &&
+	find -name \*.deb -exec dpkg -x {} "${D}" \;
+}
+#****
+
--- cygport-0.25.0/data/sample.cygport
+++ cygport-0.25.0/data/sample.cygport
@@ -6,7 +6,7 @@ SUMMARY="Cygwin source packaging tool"
 DESCRIPTION="cygport is the standard method for building and maintaining
 packages for the Cygwin distribution."
 HOMEPAGE="http://sourceware.org/cygwinports/"
-SRC_URI="https://github.com/cygwinports/cygport/${VERSION}/cygport-${VERSION}.tar.gz"
+SRC_URI="https://github.com/cygwinports/${NAME}/archive/${VERSION}.tar.gz#/${NAME}-${VERSION}.tar.gz"
 
 # this package contains no compiled Cygwin binaries
 # REMOVE THE FOLLOWING LINE for packages which are to be compiled for each arch
@@ -14,6 +14,7 @@ ARCH=noarch
 
 # Build dependencies only
 DEPEND="groff help2man robodoc"
+
 # runtime deps to go in setup.hint, and note the escaped newline
 REQUIRES="autoconf automake binutils bzip2 coreutils diffstat diffutils \
 dos2unix findutils file gawk gcc-core gcc-g++ grep gzip lftp libtool lndir \
--- cygport-0.25.0/doc/Makefile.am
+++ cygport-0.25.0/doc/Makefile.am
@@ -1,9 +1,9 @@
+if HAVE_HELP2MAN
+
 noinst_DATA = doc/manual/cygport.1.html doc/manual/index.html
 
 man_MANS = doc/cygport.1
 
-if HAVE_HELP2MAN
-
 doc/cygport.1: bin/cygport-inplace $(top_srcdir)/lib/help.cygpart
 	$(AM_V_GEN)test -d doc || $(MKDIR_P) doc ; \
 	$(HELP2MAN) -o $@ --no-info --name="Cygwin source packaging tool" $<
