Index: tar/debian/rules
===================================================================
--- tar/debian/rules
+++ tar/debian/rules
@@ -21,8 +21,6 @@ endif
 configure: configure-stamp
 configure-stamp:
 	dh_testdir
-	cp -f /usr/share/misc/config.sub build-aux/config.sub
-	cp -f /usr/share/misc/config.guess build-aux/config.guess
 	autoreconf --force -v --install
 	RSH="/usr/bin/rsh" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" ./configure \
 		--prefix=/usr \
